src/Faker/Provider/ar_SA/Company.php

```diff
@@ -60,4 +62,16 @@ public function bs()
 
         return join($result, ' ');
     }
+
+    /**
+     * example 7001010101
+     **/
+    public static function companyIdNumber()
+    {
+        do {
+            $number = static::numerify('700#######');
```

Comment:
Actually this is exactly what is done on the Person provider for the same locale, but I will look for something.